Sources/TokamakCore/Styles/ToggleStyle.swift

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Jed Fox on 07/04/2020.
+//
+// swiftlint:disable line_length
+//  Adapted from https://github.com/SwiftWebUI/SwiftWebUI/blob/16b84d46/Sources/SwiftWebUI/Views/Forms/Toggle.swift
+// swiftlint:enable line_length
+//
+
+// NOTE: ToggleStyleConfiguration.label is supposed to be a special Never View.
+// It seems like during the rendering process it’s dynamically replaced with the actual label.
+// That’s complicated so instead we’re providing the label view directly.
+
+public struct ToggleStyleConfiguration {
+  public let label: AnyView
+  @Binding public var isOn: Swift.Bool
+}
+
+public protocol ToggleStyle {
+  associatedtype Body: View
+
+  func makeBody(configuration: Self.Configuration) -> Self.Body
+
+  typealias Configuration = ToggleStyleConfiguration
+}
+
+public struct AnyToggleStyle: ToggleStyle {
```

Comment:
**nitpick**: this isn't a part of the public SwiftUI API, so under the "package internal" rules I think it should be underscored

```suggestion
public struct _AnyToggleStyle: ToggleStyle {
```